jenkins/run-server-func-tests

```diff
@@ -21,11 +21,19 @@ elif [[ -n "${1}" ]]; then
     exit 2
 fi
 
+function dump_journal {
+    printf -- "+++ journalctl dump +++\n"
+    # Try to capture the functional test container's logs.
+    podman exec ${PB_SERVER_CONTAINER_NAME} journalctl
+    printf -- "\n--- journalctl dump ---\n\n"
+}
+
 function cleanup {
     if [[ -n "${cleanup_flag}" ]]; then
         # Remove the Pbench Server container and the dependencies pod which we
         # just created and ran; remove any dangling containers; and then remove
         # any dangling images.
+        dump_journal
```

Comment:
There are several `exit` calls before we run the functional tests; I needed to see what happened inside the container. The `trap` is installed immediately before `run-pbench-in-a-can`, quite deliberately, as it creates the containers. Any failure from that point on could require analysis of the logs inside the container, which we're otherwise throwing away.

Perhaps I should have moved the *other* invocation down into the "no cleanup" exit block, though...

And, if we wanted to be fancy, we should try to conditionalize this on whether the container is running. I didn't try to get fancy.